app/controllers/application_controller.rb

```diff
@@ -288,21 +289,18 @@ def data_included?(data_type, data_id, data)
     data_id.in?(data[data_type])
   end
 
-  # rubocop:disable MethodLength
+  # TODO: refactor calling method to pass a hash to start with
   def top_three_focus(data)
-    result, freq = {}
-    3.times do
-      freq = data.each_with_object(Hash.new(0)) { |v, h| h[v] += 1 }
-      break if freq.empty?
-      max = data.max_by { |v| freq[v] }
-      break if freq[max].zero?
-      result[max] = freq[max]
-      freq.delete(max)
-      data.delete(max)
+    # Turn data array into hash of value => freq_of_value
+    freq = data.each_with_object(Hash.new(0)) do |value, hash|
+      hash[value] += 1
     end
-    result
+
+    # Sort hash and take highest 3 values
+    freq.sort_by do |occurances, _value|
```

Comment:
Change occurances => occurrences

Reply:
Thanks. I'm a terrible speller :)

Reply:
No worries! I use a spell checker in my IDE, that might be helpful